include/boost/hana/experimental/printable.hpp

```diff
@@ -111,6 +111,18 @@ BOOST_HANA_NAMESPACE_BEGIN namespace experimental {
         }
     };
 
+    // model for hana::optional
+    template <>
+    struct print_impl<hana::optional_tag> {
+        template <typename O>
+        static auto apply(O const& optional) {
```

Comment:
Perhaps they should all specify `std::string` as a return type, especially print_t::(). (I see it is specified in the doxygen version).